mitodl/ocw-studio
# Sync all files in Google Drive

#### Pre-Flight checklist
 

#### What are the relevant tickets?
- Implements backend for #578 

#### What's this PR do?
- New celery tasks to import non-video files on Google Drive and auto-create WebsiteContent resources for them, including a scheduled task to import all recent non-video files, that runs every hour by default.
- A new view that will trigger a task to import the non-video files for a specific website
- Refactor some tasks that were being called from other tasks without `raise self.replace` and/or grouping/chaining.

#### How should this be manually tested?
- Set `DRIVE_SERVICE_ACCOUNT_CREDS` and `DRIVE_SHARED_ID` with credentials from RC
- Set `DRIVE_VIDEO_UPLOADS_PARENT_FOLDER_ID=1H4HCvbmY7v5YZFeqSlbCI1TFC5MXTMY4` in your .env file.
- Create a new site with starter whose slug equals the value of your `OCW_IMPORT_STARTER_SLUG` value
- Check the google drive folder to make sure there is a new folder with name == `website.short_id` and 2 subfolders, "video_final" and "files_final"
- Repeat the instructions for importing videos (#484), but place the drive videos under the "video_final" folder - other than that it should work as before.
- Place one or more files in the "files_final" folder (images, pdfs, etc)
- Go to `http://localhost:8043/api/websites/<your_site_name>/content/gdrive_sync` and click the `POST` button.
- Very shortly you should have new resources for all the files in that folder.  The resource type ("Document", "Image", etc) for each should be a reasonable guess based on the S3 content type or file extension.  The title should equal the original filename.
- Save some changes to the new resources (for instance, change the titles).
- Add another 1 or 2 files in the drive folder.
- Go to `http://localhost:8043/api/websites/<your_site_name>/content/gdrive_sync` and click the `POST` button again.
- Check that there are again new resources for these files, the old resources still exist with the changes you made to the titles still intact, and there are no dupes. 


#### Any background context you want to provide?
- A frontend button to trigger the drive sync will be added in a future PR.
- Videos still do not have resources automatically created.  This may change in a future PR.
